src/nos/core/orchestrator.py

```diff
@@ -184,3 +188,134 @@ async def process_file_registered_notification(self, *, file_id: str):
+        )
+
+    async def _iva_code_transmitted(self, *, user: User):
+        """Send a notification that an IVA code has been transmitted to the user."""
+        await self._notification_emitter.notify(
+            email=user.email,
+            full_name=user.name,
+            notification=notifications.IVA_CODE_TRANSMITTED_TO_USER,
+        )
+
+    async def _iva_code_submitted(self, *, user: User):
+        """Send a notification to the data steward that an IVA code has been submitted."""
+        await self._notification_emitter.notify(
+            email=self._config.central_data_stewardship_email,
+            full_name=DATA_STEWARD_NAME,
+            notification=notifications.IVA_CODE_SUBMITTED_TO_DS.formatted(
+                full_user_name=user.name, email=user.email
+            ),
+        )
+
+    async def _iva_unverified(
+        self,
+        *,
+        iva_type: str,
+        user: User,
+    ):
+        """Send notifications for IVAs set to 'unverified'.
+
+        This happens when the user exceeds the allotted time to submit their IVA code.
+        """
+        await self._notification_emitter.notify(
+            email=self._config.central_data_stewardship_email,
+            full_name=DATA_STEWARD_NAME,
+            notification=notifications.IVA_UNVERIFIED_TO_DS.formatted(
+                full_user_name=user.name, email=user.email, type=iva_type
+            ),
+        )
+
+    async def process_all_ivas_invalidated(self, *, user_id: str):
+        """Send a notification to the user when all their IVAs are reset."""
+        try:
+            user = await self._user_dao.get_by_id(user_id)
+        except ResourceNotFoundError as err:
+            error = self.MissingUserError(
+                user_id=user_id, notification_name="All IVAs Invalidated"
+            )
+            log.error(
+                error,
+                extra={"user_id": user_id, "notification_name": "All IVAs Invalidated"},
+            )
+            raise error from err
+
+        await self._notification_emitter.notify(
+            email=user.email,
+            full_name=user.name,
+            notification=notifications.ALL_IVAS_INVALIDATED_TO_USER.formatted(
+                email=self._config.central_data_stewardship_email
+            ),
+        )
+
+    async def process_iva_state_change(self, *, user_iva: event_schemas.UserIvaState):
+        """Handle notifications for IVA state changes."""
+        method_map: dict[event_schemas.IvaState, Callable] = {
+            event_schemas.IvaState.CODE_REQUESTED: self._iva_code_requested,
+            event_schemas.IvaState.CODE_TRANSMITTED: self._iva_code_transmitted,
+            event_schemas.IvaState.VERIFIED: self._iva_code_submitted,
+            event_schemas.IvaState.UNVERIFIED: partial(
+                self._iva_unverified,
+                iva_type=str(user_iva.type) if user_iva.type else "N/A",
+            ),
+        }
+
+        notification_name = (
```

Comment:
I think you should create a combined dict that contains the states as keys and both the method and the notification names as values (e.g. using 2-tuples as values). Or maybe just use `method.__name__` for the error message/log instead of creating a name.

Reply:
The method.__name__ wouldn't work for the partial function so I went with the former. [c42a3cd](https://github.com/ghga-de/notification-orchestration-service/pull/5/commits/c42a3cde21b256d87535b895e0189b22d0ab506f)